app/bundles/ApiBundle/Controller/ClientController.php

```diff
@@ -68,7 +67,8 @@ public function indexAction(Request $request, PageHelperFactoryInterface $pageHe
         $request->getSession()->set('mautic.client.filter.api_mode', $apiMode);
         $request->getSession()->set('mautic.client.filter', $filter);
 
-        $clients = $this->clientModel->getEntities(
+        $model   = $this->getModel($this->getModelName());
```

Comment:
Isn't it better to use dependency injection over getting the model from a factory? The model is already injected to this class, so why do we have to get it from elsewhere? It also returns `MauticModelInterface` instead of `ClientModel` so the static analysis is missing quite a lot of valuable information.

Reply:
For the static analysis i've added some comment typehinting, which is used by IDE's and PHPStan. 

I'm not sure how to answer about injecting vs getting it from the factory. If i see this correctly the approach of getting it from the factory (through $this->getModel('name')) is used all over the controllers, so that seemed the correct / consistent approach to follow here as well. 
IMO it should be the one of the other, we use the modelFactory from the abstract class, or remove that from the abstract class and make an abstract function getModel to implement in the controllers, where they return the DI injected model. But not sure which one is best. The first (with the factory) seems to be more used ATM.

Reply:
If we'd use the injected private property then we wouldn't need the type hints in form of comments. That's more elegant solution.

These controllers were build in Symfony 2 era that used the container to get dependencies. That is no longer the suggested way. We've refactored all controllers to use DI for M5. There are still some remnants here and there that we have to iron out. 

The factory makes sense in the abstract methods where the methods don't know which model they are working with and are loading it by a service alias. We should refactor that later to use at least FQCN instead, but for now, I'd suggest to use DI wherever possible.

Reply:
Thanks for taking the time to explain. I follow the reasoning. Adjust to use back the injected clientModel.